sphinx/util/parallel.py

```diff
@@ -75,19 +79,24 @@ def __init__(self, nproc):
     def _process(self, pipe, func, arg):
         # type: (Any, Callable, Any) -> None
         try:
-            if arg is None:
-                ret = func()
-            else:
-                ret = func(arg)
-            pipe.send((False, ret))
+            logger = logging.LogCollector()
+            with logger.collect():
+                if arg is None:
+                    ret = func()
+                else:
+                    ret = func(arg)
+            failed = False
         except BaseException as err:
-            pipe.send((True, (err, traceback.format_exc())))
+            failed = True
+            ret = (err, traceback.format_exc())
+        logging.convert_serializable(logger.logs)
+        pipe.send((failed, logger.logs, ret))
```

Comment:
I feel this part is bit complicated. As a user of logging utility, it would be nice if we can write:
```
    def _process(self, pipe, func, arg):
        # type: (Any, Callable, Any) -> None
        with logging.logcollector() as collector:
            if arg is None:
                ret = func()
            else:
                ret = func(arg)
        logging.convert_serializable(collector.logs)
        pipe.send(collector.result)
```